onnxscript/rewriter/ort_fusions/cos_sin_cache.py

```diff
@@ -47,13 +43,66 @@ def __init__(
         # matched nodes as part of the rewrite-step. We apply a separate final
         # pass to remove unused nodes.
         super().__init__(name, remove_nodes=False)
-        self._max_pos_id = max_pos_id
+        # TODO: Determine what should be the default max_pos_id value
+        self._max_pos_id = 2048  # Set a default max_pos_id value
+        self._max_pos_id_from_model = False
         # map from inv_freq to (cos, sin) values for transformed graph
         self._inv_freq_cos_sin_cache: dict[ir.Value, tuple[ir.Value, ir.Value]] = {}
         self._reshape = reshape
         self._cast = cast
         self._const_freqs = const_freqs
 
+    def _set_max_position_ids_from_model(self, model: ir.Model):
```

Comment:
I think it would be better to decouple the computation of max_position_id from model from setting it here. This could be a plain setter for the _max_pos_id field. Do the computation outside ... in future, we might want to move it up elsewhere.